compiler/src/iree/compiler/Dialect/Flow/Transforms/RaiseSpecialOps.cpp

```diff
@@ -29,22 +29,28 @@ namespace Flow {
 namespace {
 
 // Method to match a transpose operation.
-static bool match2DTranspose(linalg::LinalgOp genericOp) {
+static bool matchNDTranspose(linalg::LinalgOp genericOp, unsigned rank) {
```

Comment:
This has not really checking for n-D transpose. It is checking that the inner two dimensions are transposed. Maybe rename to match that?